### ISSUE TITLE
parse panics when the matched choice is a superstring of another choice

### ISSUE THREAD

**Issue by rts-rob** [kevinmehall/rust-peg#290]
I have a grammar with the following (extremely) simplified schema:

```pest
monadicVerb = {
  "Cos" |
  "Cosh"
}
```

For context, proper invocations in the host language look as follows:
```js
  Cos(3.14159)
  // or
  Cosh(0)
```

Attempting to parse input with `parse()` fails when the input contains `Cosh(...)` but _not_ when input contains `Cos(...)`. This occurs when providing the same arguments and completely simplified to the standalone statements with a single constant argument.

This behavior also occurs with all other verbs in our grammar that are superstrings of another verb:

* `Collections` => `Collection`
* `ContainsStrRegex` => `ContainsStr`
* `Databases` => `Database`
* `FindStrRegex` => `FindStr`
* `Functions` => `Function`
* `GTE` => `GT`
* `Indexes` => `Index`
* `LTE` => `LT`
* `Logout` => `Log`
* `Roles` => `Role`
* `Singleton`/`Sinh` => `Sin`
* `Tanh` => `Tan`

I'm matching against the opening parenthesis as a workaround, e.g., `Cosh(` instead of `Cos`, so I'm not blocked.

**Comment by Mingun** [comment]
[PEG](https://en.wikipedia.org/wiki/Parsing_expression_grammar) operator `/` (choice) is _ordered_ which means that order in which you define your alternatives _is_ matter. Superstrings should always be defined before substrings.

**Comment by rts-rob** [comment]
@Mingun @kevinmehall I should have mentioned that it _was failing_ when I defined all the superstrings before all the substrings as well, e.g., `Cosh` before `Cos`. 

I say "was failing" because even though I saved that output now I cannot reproduce. 😮‍💨 

May well be some sort of dependency caching issue on my machine. Appreciate the quick responses; I'm marking as closed since I cannot reproduce now. Will back out the workaround and reopen if it pops back up.
